src/pow.rs

```diff
@@ -0,0 +1,1314 @@
+            #[doc = concat!("let a = ", stringify!($ty), "::from(0xdeadbeef);")]
+            #[doc = concat!("let b = ", stringify!($ty), "::from_be([0, 0, 0, 0xdeadbeef]);")]
+            /// assert_eq!(a, b)
+            /// ```
+            #[inline]
+            pub fn from_be(array: [u64; 4]) -> $ty { $ty(Uint256::from_be((array))) }
+
+            #[doc = concat!("Creates a [`", stringify!($ty), "`] from a little-endian array.")]
+            ///
+            /// # Examples
+            ///
+            /// ```
+            #[doc = concat!("use bitcoin::", stringify!($ty), ";")]
+            #[doc = concat!("let a = ", stringify!($ty), "::from(0xdeadbeef);")]
+            #[doc = concat!("let b = ", stringify!($ty), "::from_le([0xdeadbeef, 0, 0, 0]);")]
+            /// assert_eq!(a, b)
+            /// ```
+            #[doc = concat!("", stringify!($ty), ";")]
+            #[inline]
+            pub fn from_le(array: [u64; 4]) -> $ty { $ty(Uint256::from_le((array))) }
+
+            #[doc = concat!("Creates a [`", stringify!($ty), "`] from a big-endian byte array.")]
+            #[inline]
+            pub fn from_be_bytes(bytes: [u8; 32]) -> $ty { $ty(Uint256::from_be_bytes((bytes))) }
+
+            #[doc = concat!("Creates a [`", stringify!($ty), "`] from a little-endian byte array.")]
+            #[inline]
+            pub fn from_le_bytes(bytes: [u8; 32]) -> $ty { $ty(Uint256::from_le_bytes((bytes))) }
+
+            #[doc = concat!("Creates a [`", stringify!($ty), "`] from a big-endian byte slice.")]
+            #[inline]
+            pub fn from_be_slice(bytes: &[u8]) -> Result<$ty, ParseLengthError> {
+                let u = Uint256::from_be_slice(bytes)?;
+                Ok($ty(u))
+            }
+
+            #[doc = concat!("Creates a [`", stringify!($ty), "`] from a little-endian byte slice.")]
+            #[inline]
+            pub fn from_le_slice(bytes: &[u8]) -> Result<$ty, ParseLengthError> {
+                let u = Uint256::from_le_slice(bytes)?;
+                Ok($ty(u))
+            }
+
+            /// Converts `self` to a big-endian array.
+            ///
+            /// # Examples
+            ///
+            /// ```
+            #[doc = concat!("use bitcoin::", stringify!($ty), ";")]
+            #[doc = concat!("let a = ", stringify!($ty), "::from(0xdeadbeef);")]
+            /// assert_eq!(a.to_be(), [0, 0, 0, 0xdeadbeef]);
+            /// ```
+            #[inline]
+            pub fn to_be(self) -> [u64; 4] { self.0.to_be() }
+
+            /// Converts `self` to a little-endian array.
+            ///
+            /// # Examples
+            ///
+            /// ```
+            #[doc = concat!("use bitcoin::", stringify!($ty), ";")]
+            #[doc = concat!("let a = ", stringify!($ty), "::from(0xdeadbeef);")]
+            /// assert_eq!(a.to_le(), [0xdeadbeef, 0, 0, 0]);
+            /// ```
+            #[inline]
+            pub fn to_le(self) -> [u64; 4] { self.0.to_le() }
```

Comment:
Removed `to_le` and `to_be`. You are implying here that `from_le` and `from_be` are ok, right? Was the old name `from_array` better, or perhaps `from_be_array` (same for 'le' obviously)?

Reply:
> You are implying here that `from_le` and `from_be` are ok

No I didn't mean to imply that. Those leak the representation too.

If we're going to have them there should be no preference: `[u8; 32]`, `[u16; 16]`, `[u32; 8]`, `[u64; 4]`, `[u128; 2]` should all work because having a preference leaks representation. (apart from bytes which are just for en/decoding)